build/pmm-client-docker/entrypoint.sh

```diff
@@ -35,54 +35,143 @@ wait_for_port() {
     exit 1
 }
 
-if [ -z "${PMM_SERVER}" ]; then
-    echo PMM_SERVER is not specified. exiting
-    exit 1
-fi
-if [ -n "${PMM_USER}" ]; then
-    ARGS+=" --server-user ${PMM_USER}"
-fi
-if [ -n "${PMM_PASSWORD}" ]; then
-    ARGS+=" --server-password ${PMM_PASSWORD}"
-fi
-
-PMM_SERVER_IP=$(ping -c 1 "${PMM_SERVER/:*/}" | grep PING | sed -e 's/).*//; s/.*(//')
-SRC_ADDR=$(ip route get "${PMM_SERVER_IP}" | grep 'src ' | sed -e 's/.* src //; s/ .*//')
-CLIENT_NAME=${DB_HOST:-$HOSTNAME}
-
-wait_for_url "https://${PMM_USER}:${PMM_PASSWORD}@${PMM_SERVER}/v1/status/leader" "127.0.0.1:8300"
-
-pmm-admin config \
-    --skip-root \
+pmm2_start() {
+    AGENT_CONFIG_FILE=/usr/local/percona/pmm2/config/pmm-agent.yaml
+    rm -f ${AGENT_CONFIG_FILE}
+    if [ -n "${PMM_USER}" ]; then
+        ARGS+=" --server-username=${PMM_USER}"
+    fi
+
+    pmm-agent setup \
     --force \
-    --server "${PMM_SERVER}" \
-    --server-insecure-ssl \
-    --bind-address "${SRC_ADDR}" \
-    --client-address "${SRC_ADDR}" \
-    --client-name "${CLIENT_NAME}" \
-    ${ARGS}
-
-if [ -n "${DB_HOST}" ]; then
-    DB_ARGS+=" --host ${DB_HOST}"
-fi
-if [ -n "${DB_USER}" ]; then
-    DB_ARGS+=" --user ${DB_USER}"
-fi
-if [ -n "${DB_PASSWORD}" ]; then
-    DB_ARGS+=" --password ${DB_PASSWORD}"
-fi
-if [ -n "${DB_PORT}" ]; then
-    DB_ARGS+=" --port ${DB_PORT}"
-fi
-
-if [ -n "${DB_HOST}" -a "${DB_PORT}" ]; then
-    wait_for_port "${DB_HOST}" "${DB_PORT}"
-fi
-
-if [ -n "${DB_TYPE}" ]; then
-    pmm-admin add "${DB_TYPE}" \
+    --config-file=${AGENT_CONFIG_FILE} \
+    --server-address=${PMM_SERVER} \
+    --server-insecure-tls \
+    --container-id=${HOSTNAME} \
+    --container-name=${CLIENT_NAME} \
+    --ports-min=${CLIENT_PORT_MIN:-30100} \
+    --ports-max=${CLIENT_PORT_MAX:-30200} \
+    --listen-port=${CLIENT_PORT_LISTEN:-7777} \
+    ${ARGS} \
+    ${CLIENT_ADDR:-$SRC_ADDR} container ${CLIENT_NAME}
+
+    if [ -n "${DB_USER}" ]; then
+        DB_ARGS+=" --username=${DB_USER}"
+    fi
+    if [ -n "${DB_PASSWORD}" ]; then
+        DB_ARGS+=" --password=${DB_PASSWORD}"
+    fi
+    if [ -n "${DB_CLUSTER}" ]; then
+        DB_ARGS+=" --cluster=${DB_CLUSTER}"
+    fi
+
+    if [ -n "${DB_HOST}" -a "${DB_PORT}" ]; then
+        wait_for_port "${DB_HOST}" "${DB_PORT}"
+    fi
+
+    cat ${AGENT_CONFIG_FILE}
+
+    pmm-agent --config-file=${AGENT_CONFIG_FILE} \
+    --ports-min=${CLIENT_PORT_MIN:-30100} \
+    --ports-max=${CLIENT_PORT_MAX:-30200} > /usr/local/percona/pmm2/pmm-agent-tmp.log 2>&1 &
+
+    wait_for_url "http://127.0.0.1:7777"
+
+    cat /usr/local/percona/pmm2/pmm-agent-tmp.log
+
+    if [ -n "${DB_TYPE}" ]; then
+        case "${DB_TYPE}" in
+            mongodb )
+                pmm-admin add "${DB_TYPE}" \
+                            --skip-connection-check \
+                            --username "${MONGODB_USER}" \
+                            --password "${MONGODB_PASSWORD}" \
```

Comment:
username/password passed via DB_ARGS
```suggestion
                            ${DB_ARGS} \
```

Reply:
done